src/core/components/object.js

```diff
@@ -223,64 +231,47 @@ module.exports = function object (self) {
           const blockSize = serialized.length
           const linkLength = node.links.reduce((a, l) => a + l.size, 0)
 
-          node.toJSON((err, nodeJSON) => {
-            if (err) {
-              return callback(err)
-            }
+          const nodeJSON = node.toJSON()
 
-            callback(null, {
-              Hash: nodeJSON.Hash,
-              NumLinks: node.links.length,
-              BlockSize: blockSize,
-              LinksSize: blockSize - node.data.length,
-              DataSize: node.data.length,
-              CumulativeSize: blockSize + linkLength
-            })
+          callback(null, {
+            Hash: nodeJSON.multihash,
+            NumLinks: node.links.length,
+            BlockSize: blockSize,
+            LinksSize: blockSize - node.data.length,
+            DataSize: node.data.length,
+            CumulativeSize: blockSize + linkLength
           })
         })
       })
     }),
 
     patch: promisify({
       addLink (multihash, link, options, callback) {
-        editAndSave((node) => {
-          node.addRawLink(link)
-          return node
+        editAndSave((node, cb) => {
+          DAGNode.addLink(node, link, cb)
         })(multihash, options, callback)
       },
 
       rmLink (multihash, linkRef, options, callback) {
-        editAndSave((node) => {
-          node.links = node.links.filter((link) => {
-            if (typeof linkRef === 'string') {
-              return link.name !== linkRef
-            }
-
-            if (Buffer.isBuffer(linkRef)) {
-              return !link.hash.equals(linkRef)
-            }
-
-            if (linkRef.name) {
-              return link.name !== linkRef.name
-            }
-
-            return !link.hash.equals(linkRef.hash)
```

Comment:
It was duplicated logic, the addLink/rmLink operations on DAGNode are more complete now.